Services/CO.CDP.Organisation.WebApi/UseCase/LookupOrganisationUseCase.cs

```diff
@@ -3,11 +3,58 @@
 using CO.CDP.OrganisationInformation.Persistence;
 
 namespace CO.CDP.Organisation.WebApi.UseCase;
-public class LookupOrganisationUseCase(IOrganisationRepository organisationRepository, IMapper mapper) : IUseCase<string, Model.Organisation?>
+public class LookupOrganisationUseCase(IOrganisationRepository organisationRepository, IMapper mapper) : IUseCase<OrganisationQuery, Model.Organisation?>
 {
-    public async Task<Model.Organisation?> Execute(string name)
+    public async Task<Model.Organisation?> Execute(OrganisationQuery query)
     {
-        return await organisationRepository.FindByName(name)
-            .AndThen(mapper.Map<Model.Organisation>);
+        if (!query.IsValid)
+        {
+            throw new ArgumentException("Organisation name or identifier must be provided.");
+        }
+
+        if (!string.IsNullOrEmpty(query.Identifier))
```

Comment:
I think you're missing the test(s) to cover validation. This case is probably already covered by the previous check to `if (!query.IsValid)`.

Reply:
Thinking about it more, I might know why you've done it this way. Type checker will complain otherwise. However, this makes `IsValid` check redundant.

Wouldn't it be better if we asked for a valid identifier and a valid name?

For example:

```csharp
public async Task<Model.Organisation?> Execute(OrganisationQuery query)
{
    if (query.TryGetIdentifier(out var scheme, out var id))
    {
          return await organisationRepository.FindByIdentifier(scheme, id)
                                 .AndThen(mapper.Map<Model.Organisation>);
    }

    if (query.TryGetName(out var name))
    {
          return await organisationRepository.FindByName(query.Name)
                                 .AndThen(mapper.Map<Model.Organisation>);
    }

    throw new InvalidQueryException("Both name and identifier are missing from the request.");
}
```

Reply:
Hi Kuba, Updated this.